src/Engine/Image.php

```diff
@@ -0,0 +1,76 @@
+<?php
+declare(strict_types = 1);
+
+namespace App\Engine;
+
+use Imagine\Filter\Basic\Crop;
+use Imagine\Image\Box;
+use Imagine\Image\Point;
+
+class Image
+{
+
+    /** @var string Original URL of image. */
+    private $imageUrl;
+
+    /** @var int[] Array of floats with keys: `x`, `y`, `width`, and `height`. */
+    private $crop;
+
+    /** @var string Image data. */
+    private $data;
+
+    /** @var int Image size in bytes. */
```

Comment:
int|null

Reply:
By design, phan is quite lenient with union types. For instance, if X can have type S|T and doing some operation requires type S, phan will not complain about X possibly being a T (at least, that's how it used to work); the idea behind that is to have a low false positive rate while still being helpful. So I guess it doesn't even have a check for missing null in uninitialized class members, as it may increase the FP rate.